x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/privileged_user_monitoring_onboarding/hooks/use_fetch_privileged_user_indices.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+import { useEntityAnalyticsRoutes } from '../../../api/api';
+
+export const useFetchPrivilegedUserIndices = (query: string | undefined) => {
+  const { searchPrivMonIndices } = useEntityAnalyticsRoutes();
+  return useQuery(
+    ['POST', 'SEARCH_PRIVILEGED_USER_MONITORING_INDICES', query],
+    ({ signal }) => searchPrivMonIndices({ signal, query }),
+    {
+      keepPreviousData: true,
+      cacheTime: 0,
```

Comment:
might be worth a comment as to why you've decided not to cache?

Reply:
or does this mean cache forever I'm confused actually 😓

Reply:
I will make this comment:
`// Do not cache the data because it is used by an autocomplete query`

I think it doesn't make much sense to cache partial search results.
Imagine if you are typing `elasticsearch`. Depending on your typing speed we could cache several results, like:
`elas`,  `elastic`, `elasticsea` and `elasticsearch`

Reply:
I agree, thanks for the explanation!